subprojects/docs/src/docs/userguide/build_cache.adoc

```diff
@@ -225,6 +229,24 @@ include::sample[dir="buildCache/cacheable-bundle/groovy",files="build.gradle[tag
 include::sample[dir="buildCache/cacheable-bundle/kotlin",files="build.gradle.kts[tags=bundle-task]"]
 ====
 
+=== Using annotations
+
+Instead of using the runtime API, it is actually better to create a custom task class which is cacheable using <<more_about_tasks.adoc#sec:task_input_output_annotations,annotations to declare the inputs and outputs>>.
```

Comment:
I don't think we should say "it's actually better" if we just recommended the runtime API above without saying that it's not the best way first. Why don't we start with subclassing instead?